npm.el

```diff
@@ -201,5 +202,48 @@
     (start-process "npm-version" "*npm*" "npm" "version" version))
   )
 
+(defvar npm-node-error-regexp
+  "^[  ]+at \\(?:[^\(\n]+ \(\\)?\\([a-zA-Z\.0-9_/-]+\\):\\([0-9]+\\):\\([0-9]+\\)\)?$"
+  "Regular expression to match NodeJS errors.
+From http://benhollis.net/blog/2015/12/20/nodejs-stack-traces-in-emacs-compilation-mode/")
+
+(defvar npm-node-error-regexp-alist
+  `((,npm-node-error-regexp 1 2 3)))
+
+(defun npm-compilation-filter ()
+  "Filter function for compilation output."
+  (ansi-color-apply-on-region compilation-filter-start (point-max)))
+
+(define-compilation-mode npm-compilation-mode "Npm"
```

Comment:
This creates a custom compilation mode to run node/npm processes in. The biggest thing it adds is the error regex above that will match the node error format and let you jump to the files in the stack trace.

I was debating whether to change the existing `npm-test` function you have to use this, or whether to remove that command entirely since test comes up in the list of run commands. 

I decided to defer either option and get feedback from you.